tests/server/test_databaseservice.py

```diff
@@ -40,3 +44,56 @@ async def _wait_until_expire_is_finished():
     # Assert cleanup
     result = await data.AgentProcess.get_list()
     assert len(result) == 1
+
+
+@pytest.fixture(scope="function")
+async def set_pool_size_to_one():
+    """
+    Sets the database pool's max and min size to 1 and resets their prior values upon exiting.
+    """
+    save_min_size = opt.db_connection_pool_min_size.get()
+    save_max_size = opt.db_connection_pool_max_size.get()
+
+    opt.db_connection_pool_min_size.set("1")
+    opt.db_connection_pool_max_size.set("1")
+
+    yield
+
+    opt.db_connection_pool_min_size.set(str(save_min_size))
+    opt.db_connection_pool_max_size.set(str(save_max_size))
+
+
+@pytest.mark.slowtest
+async def test_pool_exhaustion_watcher(set_pool_size_to_one, server, caplog):
+    """
+    Test the basic functionalities of the ExhaustedPoolWatcher class
+    """
+    with caplog.at_level(logging.WARNING, logging.getLogger("inmanta.server.services.databaseservice").name):
+        database_slice = server.get_slice(databaseservice.SLICE_DATABASE)
+
+        assert database_slice._db_pool_watcher._exhausted_pool_events_count == 0
+
+        pool = database_slice._pool
+        assert pool is not None
+        connection: Connection = await pool.acquire()
+        try:
+            # Sleep long enough to make sure _check_database_pool_exhaustion gets called (scheduled to run every 200ms)
+            await asyncio.sleep(1)
+        finally:
+            await connection.close()
+
+        n_events: int = database_slice._db_pool_watcher._exhausted_pool_events_count
+        assert n_events > 0
+
+        # Call _report_database_pool_exhaustion manually (scheduled to run every 24h)
+        await database_slice._report_database_pool_exhaustion()
+
+        # Check that _report_database_pool_exhaustion resets the counter:
+        assert database_slice._db_pool_watcher._exhausted_pool_events_count == 0
```

Comment:
This one has to go as well, some new events might get recorded between this check and the call to report just above